src/PlayerOptions.cpp

```diff
@@ -1104,44 +1122,69 @@ bool PlayerOptions::FromOneModString( const RString &sOneMod, RString &sErrorOut
 	{
 	    if (sBit.find("x") != sBit.npos)
 	    {
-		for (int i=0; i<16; i++)
-		{
-		    sMod = ssprintf( "movex%d", i+1 );
-		    if( sBit == sMod)
-		    {
-			SET_FLOAT( fMovesX[i] )
-			break;
-		    }
-		}
+			for (int i=0; i<16; i++)
+			{
+				sMod = ssprintf( "movex%d", i+1 );
+				if( sBit == sMod)
+				{
+				SET_FLOAT( fMovesX[i] )
+				break;
+				}
+			}
 	    }
 	    else if (sBit.find("y") != sBit.npos)
 	    {
-		for (int i=0; i<16; i++)
-		{
-		    sMod = ssprintf( "movey%d", i+1 );
-		    if( sBit == sMod)
-		    {
-			SET_FLOAT( fMovesY[i] )
-			break;
-		    }
-		}
+			for (int i=0; i<16; i++)
+			{
+				sMod = ssprintf( "movey%d", i+1 );
+				if( sBit == sMod)
+				{
+				SET_FLOAT( fMovesY[i] )
+				break;
+				}
+			}
 	    }
 	    else if (sBit.find("z") != sBit.npos)
 	    {
-		for (int i=0; i<16; i++)
-		{
-		    sMod = ssprintf( "movez%d", i+1 );
-		    if( sBit == sMod)
-		    {
-			SET_FLOAT( fMovesZ[i] )
-			break;
-		    }
-		}
+			for (int i=0; i<16; i++)
+			{
+				sMod = ssprintf( "movez%d", i+1 );
+				if( sBit == sMod)
+				{
+				SET_FLOAT( fMovesZ[i] )
+				break;
+				}
+			}
 	    }
 	}
 	else if( sBit == "zbuffer" )				m_bZBuffer = on;
 	else if( sBit == "cosecant" )				m_bCosecant = on;
 	else if( sBit == "visualdelay" )			m_fVisualDelay = level;
+	else if( level == 0 && disabledWindows.Compare(sBit)) // "No w1" etc.
+	{	
+		static std::map<RString, TimingWindow> nameToWindow = {
+			{"w1", TW_W1},
+			{"w2", TW_W2},
+			{"w3", TW_W3},
+			{"w4", TW_W4},
+			{"w5", TW_W5},
+		};
```

Comment:
If you call the StringToX macro for the timing windows you can use `StringConversion::FromString` to convert the string representation to the enum value.

Reply:
Done!